### ISSUE TITLE
Pearsonr reports p-value of 0 with lists of length 2

### ISSUE THREAD

**Issue by guhanrv** [scipy/scipy#11027]
My issue is about ...

pearsonr. In the language R, running cor.test on a list of 2 yields an error:

cor.test(c(0.5, 0.25), c(0.25, 0.4))

Error in cor.test.default(c(0.5, 0.25), c(0.25, 0.4)) : 
  not enough finite observations

But entering the same into pearsonr gives me an ultra-confident result (see below)

#### Reproducing code example:

```
from scipy.stats import pearsonr
pearsonr([0.5, 0.25], [0.25, 0.4])

(-1.0, 0.0)
```

#### Error message:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  ...
```

#### Scipy/Numpy/Python version information:

```
('1.2.2', '1.16.4', sys.version_info(major=2, minor=7, micro=16, releaselevel='final', serial=0))
```

**Comment by WarrenWeckesser** [comment]
> But entering the same into pearsonr gives me an ultra-confident result (see below)

This was fixed in SciPy 1.3.  When the input has length 2, `pearsonr` will report a p-value of 1:

```
In [1]: import scipy                                                            

In [2]: scipy.__version__                                                       
Out[2]: '1.3.1'

In [3]: from scipy.stats import pearsonr                                        

In [4]: pearsonr([0.5, 0.25], [0.25, 0.4])                                      
Out[4]: (-1.0, 1.0)
```
The rationale is given in last paragraph of the "Notes" section of the [docstring](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.pearsonr.html).

**Comment by WarrenWeckesser** [comment]
@guhanrv, I am closing the issue, but questions or further discussion are welcome.
